packages/kbn-unified-field-list/src/hooks/use_grouped_fields.ts

```diff
@@ -52,6 +53,8 @@ export interface GroupedFieldsResult<T extends FieldListItem> {
     fieldsExistInIndex: boolean;
     screenReaderDescriptionId?: string;
   };
+  allFields: T[] | null; // `null` is for loading indicator
+  hasNewFields: boolean;
```

Comment:
Is it being used somewhere?

Reply:
yes, it's here https://github.com/elastic/kibana/blob/75cc2355c61ac0dde3066972f11fd7b827adb2cc/x-pack/plugins/lens/public/datasources/form_based/datapanel.tsx#L253 needed needed for Lens to indicate there are new fields